model/converter/json/from_domain.go

```diff
@@ -97,19 +97,27 @@ func (fd fromDomain) convertReferences(span *model.Span, preserveParentID bool)
 		length++
 	}
 	out := make([]json.Reference, 0, length)
+	var parentRef json.Reference
 	if span.ParentSpanID != 0 && !preserveParentID {
-		out = append(out, json.Reference{
+		// TODO this (wrongly) assumes that the reference type is always ChildOf
+		parentRef = json.Reference{
 			RefType: json.ChildOf,
 			TraceID: json.TraceID(span.TraceID.String()),
 			SpanID:  json.SpanID(span.ParentSpanID.String()),
-		})
+		}
+		out = append(out, parentRef)
 	}
 	for _, ref := range span.References {
-		out = append(out, json.Reference{
+		// TODO if the parentRef was wrongly added as a ChildOf ref above, this could
```

Comment:
For this TODO, it might be easier to keep track of parentRef but not add it until we're done with this for loop and if the reference isn't added, add it post. This way, if the reference should be a FollowsFrom, we don't double report.

Reply:
do you want to fix it for good then?